src/hypothesis/settings.py

```diff
@@ -257,8 +259,41 @@ def __exit__(self, *args, **kwargs):
         default_context_manager = self.defaults_stack().pop()
         return default_context_manager.__exit__(*args, **kwargs)
 
-    default = DefaultSettings()
+    @staticmethod
+    def register_profile(name, **kwargs):
+        """
+        registers a collection of values to be used as a
+        settings profile. These settings can be loaded
+        in by name. Enable different defaults for different settings
+
+        If a profile contains an invalid setting an AttributeError will be thrown
+        """
+        for setting in kwargs.keys():
+            getattr(Settings.default, setting)
+        Settings._profiles[name] = kwargs
+
+    @staticmethod
+    def _reset_defaults():
+        for name, setting in all_settings.items():
+            setattr(Settings.default, name, setting.default)
 
+    @staticmethod
+    def load_profile(name):
+        """
+        Loads in the settings defined in the profile provided
+        If the profile does not exist an AttributeError will
+        be thrown. Any setting not defined in the profile
+        will be the library defined default for that setting
+        """
+        Settings._reset_defaults()
+        try:
+            profile = Settings._profiles[name]
+        except KeyError:
+            raise ValueError("Profile '{}' has not been registered", name)
+        for key, value in profile.items():
+            setattr(Settings.default, key, value)
```

Comment:
Assigning individual settings seems like an odd choice. In particular Settings.default is assignable and it seems worth taking advantage of that.

It would be nice for example to support things that had done:

```python
@given(blah, settings=Settings.default)
def test_stuff(blah):
    ...
```

not have Settings changed underneath them.